bigchaindb/common/schema/transaction.yaml

```diff
@@ -0,0 +1,248 @@
+    - type: array
+      items:
+        "$ref": "#/definitions/base58"
+    - type: 'null'
+  sha3_hexdigest:
+    pattern: "[0-9a-f]{64}"
+    type: string
+  uuid4:
+    pattern: "[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89ab][a-f0-9]{3}-[a-f0-9]{12}"
+    type: string
+    description: |
+        A `UUID <https://en.wikipedia.org/wiki/Universally_unique_identifier>`_
+        of type 4 (random).
+  operation:
+    type: string
+    description: |
+        Type of the transaction:
+
+        A `CREATE` transaction creates an asset in BigchainDB. This
+        transaction has outputs (conditions) but no inputs (fulfillments),
+        so a dummy fulfillment is used.
+
+        A `TRANSFER` transaction transfers ownership of an asset, by providing
+        fulfillments to conditions of earlier transactions.
+
+        A `GENESIS` transaction is a special case transaction used as the
+        sole member of the first block in a BigchainDB ledger.
+    enum:
+    - CREATE
+    - TRANSFER
+    - GENESIS
+  asset:
+    type: object
+    description: |
+        Description of the asset being transacted. In the case of a `TRANSFER`
+        transaction, this field contains only the ID of asset. In the case
+        of a `CREATE` transaction, this field may contain properties:
+    additionalProperties: false
+    required:
+    - id
+    properties:
+      id:
+        "$ref": "#/definitions/uuid4"
+      divisible:
+        type: boolean
+        description: |
+            Whether or not the asset has a quantity that may be partially spent.
+      updatable:
+        type: boolean
+        description: |
+            Whether or not the description of the asset may be updated. Defaults to false.
+      refillable:
+        type: boolean
+        description: |
+          Whether the amount of the asset can change after its creation. Defaults to false.
+      data:
+        description: |
+          User provided metadata associated with the asset. May also be NULL.
+        anyOf:
+        - type: object
+          additionalProperties: true
+        - type: 'null'
+  condition:
+    type: object
+    description: |
+        An output of a transaction. A condition describes a quantity of an asset
+        and what conditions must be met in order for it to be fulfilled. See also:
+        fulfillment_.
+    additionalProperties: false
+    required:
+    - owners_after
+    - condition
+    - amount
+    properties:
+      cid:
+        type: integer
+        description: |
+            Index of this transaction's appearance in the `transaction.conditions`_
+            array. In a transaction with 2 conditions, the `cid`s will be 0 and 1.
+      condition:
+        description: |
+            Body of the condition. Has the properties:
+
+            - **details**: Details of the condition.
+            - **uri**: Condition encoded as an ASCII string.
+        type: object
+        additionalProperties: false
+        required:
+        - uri
```

Comment:
I have no memory why providing `details` is optional in `Condition`, but it shouldn't.
I'll create a ticket. Make sure that it's required for now in the schema.